gatsby-node.js

```diff
@@ -32,137 +32,31 @@ const getMainPageHandles = mainPage => {
-    const { createNodeField } = actions;
-    basePath = removeTrailingLeadingSlashes(basePath);
-    collectionPageBasePath = removeTrailingLeadingSlashes(
-      collectionPageBasePath
-    );
-
-    // Todo: Improve the way this is done. Maybe using the config.json file.
-    createNodeField({
-      node,
-      name: 'shopifyThemePath',
-      value: `${basePath && `/${basePath}`}/${collectionPageBasePath}/${
-        node.handle
-      }`,
-    });
-  }
-
-  if (node.internal.type === `ShopifyShopPolicy`) {
-    let { basePath = '', policyPageBasePath = 'policy' } = options;
-    const { createNodeField } = actions;
-    basePath = removeTrailingLeadingSlashes(basePath);
-    policyPageBasePath = removeTrailingLeadingSlashes(policyPageBasePath);
-
-    // Todo: Improve the way this is done. Maybe using the config.json file.
-    createNodeField({
-      node,
-      name: 'shopifyThemePath',
-      value: `${basePath && `/${basePath}`}/${policyPageBasePath}/${node.type}`,
-    });
-  }
-
-  if (node.internal.type === `ShopifyPage`) {
-    let { basePath = '', pageBasePath = 'pages' } = options;
-    const { createNodeField } = actions;
-    basePath = removeTrailingLeadingSlashes(basePath);
-    pageBasePath = removeTrailingLeadingSlashes(pageBasePath);
-
-    // Todo: Improve the way this is done. Maybe using the config.json file.
-    createNodeField({
-      node,
-      name: 'shopifyThemePath',
-      value: `${basePath && `/${basePath}`}${pageBasePath &&
-        `/${pageBasePath}`}/${node.handle}`,
-    });
-  }
-
-  if (node.internal.type === `ShopifyBlog`) {
-    let { basePath = '', blogPageBasePath = 'blog' } = options;
-    const { createNodeField } = actions;
-    basePath = removeTrailingLeadingSlashes(basePath);
-    if (blogPageBasePath.length > 0) {
-      blogPageBasePath = removeTrailingLeadingSlashes(blogPageBasePath);
-    }
-
-    let nodeUrlArray = node.url.split('/');
-    let blogHandle = nodeUrlArray[nodeUrlArray.length - 1];
-
-    // As while creating only new nodes we do not know about already existing
-    // We need to store information about blogs received early in cache.
-    // 1. Push new blogs to array
-    availableBlogs.push({
-      shopifyId: node.shopifyId,
-      handle: blogHandle,
-    });
-    // 2. Receive already known blogs from cache
-    let blogs = await cache.get('availableBlogs');
-    // 3. Concat new blogs with already known
-    if (blogs && blogs.length > 0) {
-      availableBlogs = availableBlogs.concat(blogs);
-    }
-    // 4. Write back to cache
-    await cache.set('availableBlogs', availableBlogs);
-
-    // Todo: Improve the way this is done. Maybe using the config.json file.
-    createNodeField({
-      node,
-      name: 'shopifyThemePath',
-      value: `${basePath && `/${basePath}`}${blogPageBasePath &&
-        `/${blogPageBasePath}`}/${blogHandle}`,
-    });
-  }
+exports.onPreInit = (_, pluginOptions) => {
+  isShopifyLite = pluginOptions.shopifyLite;
```

Comment:
saves plugin option  in global variable to be used later in `schemacustomization` and `createPages`